xarray/core/computation.py

```diff
@@ -1482,6 +1506,53 @@ def _cov_corr(
         return corr
 
 
+def _weighted_cov_corr(
+    da_a: T_DataArray,
+    da_b: T_DataArray,
+    weights: T_DataArray,
+    dim: Dims = None,
+    ddof: int = 0,
+    method: Literal["cov", "corr", None] = None,
+) -> T_DataArray:
+    """
+    Internal method for weighted xr.cov() and xr.corr(), extending
+    _cov_corr() functionality.
+    """
+    # 1. Broadcast the two arrays
+    da_a, da_b = align(da_a, da_b, join="inner", copy=False)
+
+    # 2. Ignore the nans
+    valid_values = da_a.notnull() & da_b.notnull()
+    da_a = da_a.where(valid_values)
+    da_b = da_b.where(valid_values)
+
+    # 3. Detrend along the given dim
+    demeaned_da_a = da_a - da_a.weighted(weights).mean(dim=dim)
+    demeaned_da_b = da_b - da_b.weighted(weights).mean(dim=dim)
+
+    # 4. Compute covariance along the given dim
+    # N.B. `skipna=True` is required or auto-covariance is computed incorrectly. E.g.
+    # Try xr.cov(da,da) for da = xr.DataArray([[1, 2], [1, np.nan]], dims=["x", "time"])
+    cov = (
+        (demeaned_da_a.conj() * demeaned_da_b)
```

Comment:
What's the role of this `.conj`?

Reply:
This was copy/pasted from the previous code; looks like it's there to support complex arrays